### PR TITLE
Added warning message

### PR DESCRIPTION
**Pull request by Jay-sanjay**
This PR is related to #244 

## PR CheckList

- [x]  Added warning message in `tron.jl` and `tronls.jl`

### CODE REVIEW

**Comment by Jay-sanjay** [comment]
@tmigot  Sir would you please provide workflows approval to this PR.

File: src/tron.jl [hunk 1]
```diff
@@ -217,6 +217,10 @@ function SolverCore.solve!(
     error("tron should only be called for unconstrained or bound-constrained problems")
   end
 
+  if(verbose > 0)
```
Comment (tmigot) [inline src/tron.jl:220]: Ok, but we need to show this warning if `verbose > 0` AND the initial guess provided is not within bounds.

**Review by tmigot — CHANGES_REQUESTED** [review]

**Comment by Jay-sanjay** [comment]
bump.

File: src/tron.jl [hunk 2]
```diff
@@ -217,6 +217,10 @@ function SolverCore.solve!(
     error("tron should only be called for unconstrained or bound-constrained problems")
   end
 
+  if (verbose > 0 && (ℓ > x && x > u))
```
Comment (tmigot) [inline src/tron.jl:220]:
```suggestion
  if (verbose > 0 && !(u > x > ℓ))
```
looks more compact.

File: src/tronls.jl [hunk 3]
```diff
@@ -236,6 +236,10 @@ function SolverCore.solve!(
     error("tron should only be called for unconstrained or bound-constrained problems")
   end
 
+  if(verbose > 0)
```
Comment (tmigot) [inline src/tronls.jl:239]:
```suggestion
  if (verbose > 0 && !(u > x > ℓ))
```

**Review by tmigot — CHANGES_REQUESTED** [review]
Thank you for the updated suggestion, we are getting there!
Could you also add a unit test for this using @test_warn? https://docs.julialang.org/en/v1/stdlib/Test/#Test.@test_warn
This will improve coverage. Thanks !!

**Review by tmigot — APPROVED** [review]
@Jay-sanjay I took the liberty to complete your PR. Thanks for your help.

**Comment by Jay-sanjay** [comment]
Thanks @tmigot  for it, got a bit busy with my semester exams so could not find proper time for the PR ;) 
That said thanks for the help :)

**Pull request merged**